code/src/cljs/sixsq/nuvla/ui/profile/events.cljs

```diff
@@ -459,3 +461,77 @@
   ::set-active-tab-index
   (fn [db [_ active-tab-index]]
     (assoc db ::spec/active-tab-index active-tab-index)))
+
+
+(reg-event-fx
+  ::code-validation-2fa-failed
+  (fn [_ [_ response]]
+    {:fx [[:dispatch [::set-error (-> response :response :message)]]]}))
+
+(reg-event-fx
+  ::code-validation-2fa-success
+  (fn [_ [_ success-header success-content]]
+    {:fx [[:dispatch [::close-modal]]
+          [:dispatch [::messages-events/add
+                      {:header  success-header
+                       :content success-content
+                       :type    :success}]]
+          [:dispatch [::get-user]]]}))
+
+
+(reg-event-fx
+  ::validate-2fa-activation
+  (fn [{{:keys [::spec/callback-2fa]} :db} [_ token success-header success-content]]
+    {:http-xhrio {:method          :put
+                  :uri             callback-2fa
+                  :format          (ajax/json-request-format)
+                  :params          {:token token}
+                  :response-format (ajax/json-response-format {:keywords? true})
+                  :on-success      [::code-validation-2fa-success success-header success-content]
+                  :on-failure      [::code-validation-2fa-failed]}}))
+
+
+(reg-event-db
+  ::set-callback-2fa
+  (fn [{:keys [::spec/user] :as db} [_ response loading-key]]
+    (let [callback-url (:location response)]
+      (-> db
+          (update ::spec/loading disj loading-key)
+          (assoc ::spec/callback-2fa callback-url)))))
+
+
+(reg-event-fx
+  ::enable-2fa
+  (fn [{{:keys [::spec/user] :as db} :db}]
+    {:db                     (-> db
+                                 (update ::spec/loading conj :enable-2fa)
+                                 (assoc ::spec/open-modal :enable-2fa)
+                                 (assoc ::spec/callback-2fa nil))
+     ::cimi-api-fx/operation [(:id user) "enable-2fa"
```

Comment:
the email will have the same body as your typical sign up email, which means it will say: "If you didn't initiate this request, do NOT click on the **link** and report this to the service administrator.". But there's no link to click on.